Upgraded tests #977

mlesnic opened the pull request:
So basically it is not finished yet (i didn't removed current timeout approach in ConnectivityMonitorTest) but because I will be off until next wendesday, I am creating this PR to run this tests in CI and also for partial review. There are still some Thread.sleeps but there are in reasonable places, for example the timeout for executing current operation in StartStopOperationHandler is hardcoded in field "OPERATION_TIMEOUT_IN_MILLISECONDS", so I can rely on it.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/982)
<!-- Reviewable:end -->

ThaliTester commented:
PR is added to the queue for testing as 1. task. (b570947)

msftclas commented:
Hi __@mlesnic__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

ThaliTester commented:
Test (Fail) 83627337 build is completed (b570947)

See https://github.com/ThaliTester/TestResults/tree/83627337b570947_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (140e0c5)

ThaliTester commented:
PR is added to the queue for testing as 2. task. (a1c904e)

ThaliTester commented:
Test (Success) 83627337 build is completed (a1c904e)

See https://github.com/ThaliTester/TestResults/tree/83627337a1c904e_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337140e0c5(140e0c5) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337140e0c5_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
Test 83627337a1c904e(a1c904e) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337a1c904e_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (381d561)

ThaliTester commented:
Test (Success) 83627337 build is completed (381d561)

See https://github.com/ThaliTester/TestResults/tree/83627337381d561_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337381d561(381d561) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337381d561_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (941f206)

ThaliTester commented:
Test (Success) 83627337 build is completed (941f206)

See https://github.com/ThaliTester/TestResults/tree/83627337941f206_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337941f206(941f206) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337941f206_Upgraded_tests__977_mlesnic/ for the fail logs

yaronyg commented:
@evabishchevich and @vasilevskayaem, you two should thumb wrestle to decide which of you two should review this. I am also reviewing it but obviously no Android code should be going into the project with out at least one of your two having reviewed and approved it.

---

Reviewed 4 of 8 files at r1, 5 of 6 files at r2, 1 of 1 files at r3, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 81 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR5yMjD6eWn0Ki6lHEN:-KR5yMjD6eWn0Ki6lHEO:buwzm) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L81)):*
> ```Java
>                 while (!mConnectionHelper.getDiscoveryManager().isRunning()) {
>                     try {
>                         Thread.sleep(3000);
> ```

Can we shorten this to say 500 ms? We want tests to run as quickly as possible. Also, what happens if isRunning is never true? How is this detected? 

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 158 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR604u7MWpCprHCa1LG:-KR604u7MWpCprHCa1LH:b-kbuyz0) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L158)):*
> ```Java
>     public void testDispose() throws Exception {
>         if (!isBLESupported) {
>             assertThat("Start method returns true",
> ```

I would suggest that you can re-write this if/else without an if else and instead just put in the call to mConnectionHelper.start and is(equalTo(true)) and just put isBLESupported as the second argument to mConnectionHelper.start()

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 182 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR6-3pDV8F1qGsnOA5H:-KR6-3pElYMpda21mKdM:b-f3itqa) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L182)):*
> ```Java
>         checkDiscoveryManagerNotRunning.start();
>         checkDiscoveryManagerNotRunning.join();
>         Thread.sleep(mOperationTimeout);
> ```

If we have confirmed that the discovery manager isn't running then why do we need to wait another 3 seconds? cancelCurrentOperation is instantaneous.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 220 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR60KkKjrRQNkbLn2sb:-KR60KkLRbvAX93mtQRp:b8kf33q) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L220)):*
> ```Java
>     public void testStart() throws Exception {
>         if (!isBLESupported) {
>             assertThat("Start method returns true",
> ```

I would suggest the same trick as above where we replace the false/true with isBLESupported and then after these two lines you can have a if (isBLESupported) check to run checkDiscoveryManagerRunning

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 268 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR60bZLPrK_yNWV9K-P:-KR60bZLPrK_yNWV9K-Q:b-21dr48) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L268)):*
> ```Java
>         checkDiscoveryManagerNotRunning.start();
>         checkDiscoveryManagerNotRunning.join();
>         Thread.sleep(mOperationTimeout);
> ```

Why are we waiting for a time out? What operation is timing out?

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 299 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR60gpGAnvJMpncC4n0:-KR60gpHyt0VGuf4SWZL:b-i79wtb) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L299)):*
> ```Java
>         if (!isBLESupported) {
>             assertThat("DiscoveryManager isRunning should return false",
>                     mConnectionHelper.getDiscoveryManager().isRunning(), is(false));
> ```

Same change as before. It's just not good style to repeat two nearly identical function calls with a single arg difference.

---

*[src/android/test/io/jxcore/node/ConnectivityMonitorTest.java, line 74 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR61BVCRssGkGonF0Ux:-KR61BVCRssGkGonF0Uy:b-xx1c6u) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectivityMonitorTest.java#L74)):*
> ```Java
>             @Override
>             public void run() {
>                 while (mBluetoothAdapter.isEnabled() != currentBTState && counter < 2) {
> ```

Why 2?

---

*[src/android/test/io/jxcore/node/ConnectivityMonitorTest.java, line 76 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR60xXGRnTGFLAzXFuQ:-KR60xXHPRfiycU6jI5q:b-8v11ey) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectivityMonitorTest.java#L76)):*
> ```Java
>                 while (mBluetoothAdapter.isEnabled() != currentBTState && counter < 2) {
>                     try {
>                         Thread.sleep(3000);
> ```

Again, please, a shorter wait.

---

*[src/android/test/io/jxcore/node/IncomingSocketThreadTest.java, line 110 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR61qmMsdeifNH5MAgF:-KR61qmNazPWiwgOqnQz:belowmg) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/IncomingSocketThreadTest.java#L110)):*
> ```Java
>                 while (mOutgoingSocketThread.mServerSocket == null) {
>                     try {
>                         Thread.sleep(3000);
> ```

And if it is always null? What happens? Oh and shorter wait please. :)

---

*[src/android/test/io/jxcore/node/IncomingSocketThreadTest.java, line 131 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR61w3LMVE3Kl3lZSnT:-KR61w3LMVE3Kl3lZSnU:bpyxq8u) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/IncomingSocketThreadTest.java#L131)):*
> ```Java
>         });
> 
>         mOutgoingSocketThread.setPort(47775);
> ```

Where did these port numbers come from? And why aren't they constants?

---

*[src/android/test/io/jxcore/node/StreamCopyingThreadTest.java, line 88 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR62u7OIkPY0btIeNuN:-KR62u7OIkPY0btIeNuO:bug83lx) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/StreamCopyingThreadTest.java#L88)):*
> ```Java
>         mStreamCopyingThread.start();
> 
>         Thread.sleep(2000);
> ```

Why are we using a sleep? Why aren't we just checking that the data was copied to the output and then closing once it is?

---

*[src/android/test/io/jxcore/node/StreamCopyingThreadTest.java, line 191 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR63IgwjtmndzmEToJ7:-KR63IgxQyfUt1DQ7Tjz:bx3v56q) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/StreamCopyingThreadTest.java#L191)):*
> ```Java
> 
>             try {
>                 Thread.sleep((random.nextInt(200) + 50));
> ```

A comment explaining the purpose of this sleep would be good.

---

*[src/android/test/io/jxcore/node/StreamCopyingThreadTest.java, line 214 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR63vk17onVjfe-7VkY:-KR63vk17onVjfe-7VkZ:bci32nk) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/StreamCopyingThreadTest.java#L214)):*
> ```Java
>         @Override
>         public void write(int oneByte) throws IOException {
>             counter++;
> ```

This is just begging for problems. We should define the counter locally to this class, not on the global parent class. Otherwise someone else could mess with it to bad results.

---

*[test/www/jxcore/UnitTest_app.js, line 15 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR64E-k9CqWfaRw_Sz2:-KR64E-l6oogEqz1xAx1:bclhkuc) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/test/www/jxcore/UnitTest_app.js#L15)):*
> ```JavaScript
> var testUtils = require('./lib/testUtils');
> var ThaliMobile = require('thali/NextGeneration/thaliMobile');
> var radiosPromise = require('lie');
> ```

Please just call this 'Promise'. It's not specific to the radios. 

---

*[test/www/jxcore/UnitTest_app.js, line 46 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR64kQkksQYriBI4MgL:-KR64kQkksQYriBI4MgM:b-xxxlm3) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/test/www/jxcore/UnitTest_app.js#L46)):*
> ```JavaScript
>         });
> 
>         setImmediate(function () {
> ```

Why do we need the setImmediate? I'm not even sure if 563 (mentioned below) still repros.

---

*[test/www/jxcore/UnitTest_app.js, line 67 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR64NUrYFhnCh-zWUE-:-KR64NUrYFhnCh-zWUE0:bfmjihp) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/test/www/jxcore/UnitTest_app.js#L67)):*
> ```JavaScript
>           });
>         });
>       }, function (err) {
> ```

Please use catch, not the second function argument. What you did here is correct but from a stylistic perspective we want to make it easy for devs to know where to expect things to happen in our code so we always use catch.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-:-KR5xXi4kQZI4330ePWH:bqcs93i)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 1. task. (0e3995d)

ThaliTester commented:
Test (Fail) 83627337 build is completed (0e3995d)

See https://github.com/ThaliTester/TestResults/tree/836273370e3995d_Upgraded_tests__977_mlesnic/ for the logs

mlesnic commented:
Review status: 8 of 12 files reviewed at latest revision, 16 unresolved discussions.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 158 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR604u7MWpCprHCa1LG:-KR7hQTJbhN-MR8mTq3o:b-5loda8) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L158)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> I would suggest that you can re-write this if/else without an if else and instead just put in the call to mConnectionHelper.start and is(equalTo(true)) and just put isBLESupported as the second argument to mConnectionHelper.start()
</details>
Yes, great idea

---

*[src/android/test/io/jxcore/node/IncomingSocketThreadTest.java, line 131 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR61w3LMVE3Kl3lZSnT:-KR7jCLWFxbPKRci-NW3:biaczx2) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/IncomingSocketThreadTest.java#L131)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> Where did these port numbers come from? And why aren't they constants?
</details>
Honestly, I have no idea. But it really doesn't matter. I will make it final.

---

*[src/android/test/io/jxcore/node/StreamCopyingThreadTest.java, line 88 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR62u7OIkPY0btIeNuN:-KR7yEccnBahS8SUlCjP:b2b2si0) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/StreamCopyingThreadTest.java#L88)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> Why are we using a sleep? Why aren't we just checking that the data was copied to the output and then closing once it is?
</details>
Because the stream doesn't end, so what I do here is wait to load some data and then check if all loaded data was properly copied to the output stream. The longer I wait, the larger data is streamed.

---

*[test/www/jxcore/UnitTest_app.js, line 15 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR64E-k9CqWfaRw_Sz2:-KR7jxvBncIJNIAi4DlD:b-vywwm4) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/test/www/jxcore/UnitTest_app.js#L15)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> Please just call this 'Promise'. It's not specific to the radios. 
</details>
If I call this 'Promise', JSHint tells me that this name is reserved in future version of javascript.

---

*[test/www/jxcore/UnitTest_app.js, line 46 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR64kQkksQYriBI4MgL:-KR7v1vL38QODpxaEo91:b-1srxt6) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/test/www/jxcore/UnitTest_app.js#L46)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> Why do we need the setImmediate? I'm not even sure if 563 (mentioned below) still repros.
</details>
Because we need to wait until unit tests will end, otherwise, node tests and native tests will interfere with each other.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982)*
<!-- Sent from Reviewable.io -->

mlesnic commented:
Review status: 8 of 12 files reviewed at latest revision, 16 unresolved discussions.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 268 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR60bZLPrK_yNWV9K-P:-KR80Bs28HsgY8av2Wcu:ba3dfob) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L268)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> Why are we waiting for a time out? What operation is timing out?
</details>
Removed

---

*[src/android/test/io/jxcore/node/ConnectivityMonitorTest.java, line 74 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR61BVCRssGkGonF0Ux:-KR7zyO-4ZiMJKDSwxai:b-47r5n8) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/ConnectivityMonitorTest.java#L74)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> Why 2?
</details>
So I changed sleeps to smaller values and counter limit to 10, so phone has 5s to turn on specific radio. I think this is more than enough.

---

*[src/android/test/io/jxcore/node/StreamCopyingThreadTest.java, line 191 \[r5\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR63IgwjtmndzmEToJ7:-KR8-aNBtCh-MlifU3Ch:b-68pve9) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/941f2064bfb6eb8bd5ca4e46fb2fbd510ed8f3e8/src/android/test/io/jxcore/node/StreamCopyingThreadTest.java#L191)):*
<details><summary><i>Previously, yaronyg (Yaron Y Goland) wrote…</i></summary>
> A comment explaining the purpose of this sleep would be good.
</details>
So basically, as I understood correctly, there was a problem with streams when data arrived in different time intervals. So what I'm doing here, I send data in non constant intervals. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 1. task. (1800284)

ThaliTester commented:
Test (Fail) 83627337 build is completed (1800284)

See https://github.com/ThaliTester/TestResults/tree/836273371800284_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (84eab4b)

ThaliTester commented:
Test (Success) 83627337 build is completed (84eab4b)

See https://github.com/ThaliTester/TestResults/tree/8362733784eab4b_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 8362733784eab4b(84eab4b) has failed

See https://github.com/ThaliTester/TestResults/tree/8362733784eab4b_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (315fde5)

ThaliTester commented:
Test (Success) 83627337 build is completed (315fde5)

See https://github.com/ThaliTester/TestResults/tree/83627337315fde5_Upgraded_tests__977_mlesnic/ for the logs

vasilevskayaem commented:
Review status: 3 of 11 files reviewed at latest revision, 19 unresolved discussions.

---

*[src/android/test/com/test/thalitest/ThaliTestRunner.java, line 34 \[r8\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR8YN5XbbWwlO35CK6M:-KR8YN5YCoGmh-pf2qZv:b-tnscj1) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/84eab4b5f63f5d44960ba5b160f6f45aba22346e/src/android/test/com/test/thalitest/ThaliTestRunner.java#L34)):*
>   Thread checkRadiosThread = new Thread(new Runnable() {
<details><summary><i alt="summary">Quoted 13 lines of code…</i></summary>
>             int counter = 0;
>             @Override
>             public void run() {
>                 do {
>                     try {
>                         Thread.sleep(3000);
>                         counter++;
>                     } catch (InterruptedException e) {
>                         e.printStackTrace();
>                     }
>                 } while (!btAdapter.isEnabled() && !wifiManager.isWifiEnabled() && counter < 2);
>             }
>         });
</details>

Could you please reduce sleep interval and increase counter max value, for example like you did: sleep 500 and max counter value set to 10, declare constants for this values and reuse it in other threads if needed. Can you also make class for this thread inner like here createCheckDiscoveryManagerNotRunningThread. So this code can be reused somewhere.  Also if counter become great than  max value but !btAdapter.isEnabled() = false log some message that app wait too long but nothing happens.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 85 \[r8\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR8cBjYwSnDhr_OtkQ8:-KR8cBjYwSnDhr_OtkQ9:b-1jks1s) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/84eab4b5f63f5d44960ba5b160f6f45aba22346e/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L85)):*
> createCheckDiscoveryManagerRunningThread
Change class names to start with uppercase letter. CreateCheckDiscoveryManagerRunningThread and CreateCheckDiscoveryManagerNotRunningThread

---

*[src/android/test/io/jxcore/node/ConnectivityMonitorTest.java, line 236 \[r8\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR8hIYIkB4pxA3iM2xY:-KR8hIYJDvqJcLIeeQiW:b-qextqr) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/84eab4b5f63f5d44960ba5b160f6f45aba22346e/src/android/test/io/jxcore/node/ConnectivityMonitorTest.java#L236)):*
>   } while (!mBluetoothAdapter.isEnabled() && counter < 2);
log some message if we wait long time but bluetooth still disabled. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
Test 83627337315fde5(315fde5) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337315fde5_Upgraded_tests__977_mlesnic/ for the fail logs

mlesnic commented:
Review status: 3 of 11 files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 85 \[r8\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR8cBjYwSnDhr_OtkQ8:-KRD2TpgnEXHd4a2ijGL:b-gp80ds) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/84eab4b5f63f5d44960ba5b160f6f45aba22346e/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L85)):*
<details><summary><i>Previously, vasilevskayaem wrote…</i></summary>
> > createCheckDiscoveryManagerRunningThread
> Change class names to start with uppercase letter. CreateCheckDiscoveryManagerRunningThread and CreateCheckDiscoveryManagerNotRunningThread
</details>
These are not classes but methods.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
PR is added to the queue for testing as 2. task. (425e1a9)

vasilevskayaem commented:
Review status: 3 of 14 files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[src/android/test/io/jxcore/node/ConnectionHelperTest.java, line 85 \[r8\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982#-KR8cBjYwSnDhr_OtkQ8:-KRD9u3XVACRCjnkqFsN:b5c2eie) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/84eab4b5f63f5d44960ba5b160f6f45aba22346e/src/android/test/io/jxcore/node/ConnectionHelperTest.java#L85)):*
<details><summary><i>Previously, mlesnic (Marcin Lesniczek) wrote…</i></summary>
> These are not classes but methods.
</details>
sorry)) probably it's better to create classes for threads with check functionality, just in case if we want to reuse this classes in different tests. All this "waiting" threads has the same parts: check some condition and counter value, sleep and increase counter. And we have many repeated parts in code. You can skip this point for now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/982)*
<!-- Sent from Reviewable.io -->

ThaliTester commented:
Test (Fail) 83627337 build is completed (425e1a9)

See https://github.com/ThaliTester/TestResults/tree/83627337425e1a9_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (37d736b)

ThaliTester commented:
Test (Fail) 83627337 build is completed (37d736b)

See https://github.com/ThaliTester/TestResults/tree/8362733737d736b_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (6cb74ed)

ThaliTester commented:
Test (Fail) 83627337 build is completed (6cb74ed)

See https://github.com/ThaliTester/TestResults/tree/836273376cb74ed_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (ab51778)

ThaliTester commented:
PR is added to the queue for testing as 4. task. (5fe617f)

ThaliTester commented:
Test (Success) 83627337 build is completed (ab51778)

See https://github.com/ThaliTester/TestResults/tree/83627337ab51778_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test (Success) 83627337 build is completed (5fe617f)

See https://github.com/ThaliTester/TestResults/tree/836273375fe617f_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337ab51778(ab51778) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337ab51778_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
Test 836273375fe617f(5fe617f) has failed

See https://github.com/ThaliTester/TestResults/tree/836273375fe617f_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 8. task. (05cfec3)

ThaliTester commented:
Test (Success) 83627337 build is completed (05cfec3)

See https://github.com/ThaliTester/TestResults/tree/8362733705cfec3_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 8362733705cfec3(05cfec3) has failed

See https://github.com/ThaliTester/TestResults/tree/8362733705cfec3_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (1ec6637)

ThaliTester commented:
Test (Fail) 83627337 build is completed (1ec6637)

See https://github.com/ThaliTester/TestResults/tree/836273371ec6637_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (35e580c)

ThaliTester commented:
Test (Fail) 83627337 build is completed (35e580c)

See https://github.com/ThaliTester/TestResults/tree/8362733735e580c_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (57e450b)

ThaliTester commented:
Test (Fail) 83627337 build is completed (57e450b)

See https://github.com/ThaliTester/TestResults/tree/8362733757e450b_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (2e7ca3d)

ThaliTester commented:
Test (Success) 83627337 build is completed (2e7ca3d)

See https://github.com/ThaliTester/TestResults/tree/836273372e7ca3d_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 836273372e7ca3d(2e7ca3d) has failed

See https://github.com/ThaliTester/TestResults/tree/836273372e7ca3d_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 2. task. (00cd7fa)

ThaliTester commented:
Test (Success) 83627337 build is completed (00cd7fa)

See https://github.com/ThaliTester/TestResults/tree/8362733700cd7fa_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 8362733700cd7fa(00cd7fa) has failed

See https://github.com/ThaliTester/TestResults/tree/8362733700cd7fa_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (9690449)

ThaliTester commented:
Test (Success) 83627337 build is completed (9690449)

See https://github.com/ThaliTester/TestResults/tree/836273379690449_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 836273379690449(9690449) has failed

See https://github.com/ThaliTester/TestResults/tree/836273379690449_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (142cb7e)

ThaliTester commented:
PR is added to the queue for testing as 3. task. (7398f6e)

ThaliTester commented:
Test (Fail) 83627337 build is completed (142cb7e)

See https://github.com/ThaliTester/TestResults/tree/83627337142cb7e_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (95b1a33)

ThaliTester commented:
PR is added to the queue for testing as 4. task. (0459b7a)

ThaliTester commented:
Test (Fail) 83627337 build is completed (7398f6e)

See https://github.com/ThaliTester/TestResults/tree/836273377398f6e_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test (Success) 83627337 build is completed (95b1a33)

See https://github.com/ThaliTester/TestResults/tree/8362733795b1a33_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test (Success) 83627337 build is completed (0459b7a)

See https://github.com/ThaliTester/TestResults/tree/836273370459b7a_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 8362733795b1a33(95b1a33) has failed

See https://github.com/ThaliTester/TestResults/tree/8362733795b1a33_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
Test 836273370459b7a(0459b7a) has failed

See https://github.com/ThaliTester/TestResults/tree/836273370459b7a_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (ae40023)

ThaliTester commented:
Test (Success) 83627337 build is completed (ae40023)

See https://github.com/ThaliTester/TestResults/tree/83627337ae40023_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337ae40023(ae40023) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337ae40023_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (e0b549f)

ThaliTester commented:
Test (Success) 83627337 build is completed (e0b549f)

See https://github.com/ThaliTester/TestResults/tree/83627337e0b549f_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337e0b549f(e0b549f) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337e0b549f_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (1654557)

ThaliTester commented:
Test (Fail) 83627337 build is completed (1654557)

See https://github.com/ThaliTester/TestResults/tree/836273371654557_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (3ce2df7)

ThaliTester commented:
Test (Success) 83627337 build is completed (3ce2df7)

See https://github.com/ThaliTester/TestResults/tree/836273373ce2df7_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 836273373ce2df7(3ce2df7) has failed

See https://github.com/ThaliTester/TestResults/tree/836273373ce2df7_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 4. task. (b783869)

ThaliTester commented:
Test (Success) 83627337 build is completed (b783869)

See https://github.com/ThaliTester/TestResults/tree/83627337b783869_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337b783869(b783869) has failed

See https://github.com/ThaliTester/TestResults/tree/83627337b783869_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (7282b4e)

ThaliTester commented:
Test (Success) 83627337 build is completed (7282b4e)

See https://github.com/ThaliTester/TestResults/tree/836273377282b4e_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 836273377282b4e(7282b4e) has failed

See https://github.com/ThaliTester/TestResults/tree/836273377282b4e_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (2ac050d)

ThaliTester commented:
Test (Success) 83627337 build is completed (2ac050d)

See https://github.com/ThaliTester/TestResults/tree/836273372ac050d_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 836273372ac050d(2ac050d) has failed

See https://github.com/ThaliTester/TestResults/tree/836273372ac050d_Upgraded_tests__977_mlesnic/ for the fail logs

ThaliTester commented:
PR is added to the queue for testing as 3. task. (176b608)

ThaliTester commented:
Test (Success) 83627337 build is completed (176b608)

See https://github.com/ThaliTester/TestResults/tree/83627337176b608_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337176b608(176b608) has successfully finished without an error

See https://github.com/ThaliTester/TestResults/tree/83627337176b608_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
PR is added to the queue for testing as 1. task. (176b608)

msftclas commented:
Hi __@mlesnic__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

ThaliTester commented:
Test (Success) 83627337 build is completed (176b608)

See https://github.com/ThaliTester/TestResults/tree/83627337176b608_Upgraded_tests__977_mlesnic/ for the logs

ThaliTester commented:
Test 83627337176b608(176b608) has successfully finished without an error

See https://github.com/ThaliTester/TestResults/tree/83627337176b608_Upgraded_tests__977_mlesnic/ for the logs
